src/main/java/com/marklogic/kafka/connect/source/CsvPlanExecutor.java

```diff
@@ -0,0 +1,40 @@
+package com.marklogic.kafka.connect.source;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import org.apache.kafka.connect.source.SourceRecord;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CsvPlanExecutor implements PlanExecutor {
```

Comment:
I made this new PlanExecutor interface and copied the code of interest from each RBB to the appropriate PE impl.